testsuite/run_sets/secondary_destructive.yml

```diff
@@ -4,13 +4,10 @@
 # feature files in testsuite/docs/Guidelines.md in "Rules for features" chapter,
 # as well as guidelines about idempotency in "Idempotency" chapter.
 
+## Finishing Secondary features  BEGIN ##
 
-## Post run features BEGIN ##
+# this feature is destructive for other features, so it is better at the end
```

Comment:
"Destructive" looks like a hard word for me...
Maybe we can have two yml:
- finishing.yml
- collect_logs.yml

Reply:
I have to admit that I find the word not that hard because the server hostname rename test is indeed very destructive if it fails. Maybe it is not the proper word, but at least now, we have it in its own YAML file where everybody know it could be problematic.

Reply:
Shouldn't our tests be isolated? The outcome of one test case should not impact the execution or results of others no matter what.

Reply:
Thinking of a word which carries a similar meaning and can fit, "disruptive" comes to mind.
But, well, "destructive" does make the point pretty clear.

Reply:
I agree it's a bit violent - and it's also inexact.

What we want to state is that they are non idempotent. But they do not "destroy" anything :tongue: 

Why do we need to rename? :thinking:

Reply:
> Shouldn't our tests be isolated? The outcome of one test case should not impact the execution or results of others no matter what.

Yes, that's the theory. We need to have as close idempotent test cases as possible.

BUT, our environment setup is quite heavy, so we can't deploy a clean environment for each test, meaning we need to reuse the server, its database, its packages, and all the systems registered.

What we do, is try to roll back some actions with "Clean-Up" scenarios. But those scenarios can break, or we can fail earlier and not be able to do a proper cleanup, and then all fall in a cascade. This is something we need to live with for now.

Maybe that can improve when we move from VMs to Containers, it's something we can explore in the future.

Reply:
Ok, https://github.com/uyuni-project/uyuni/pull/8128#discussion_r1448667695

Closing this thread for now, feel free to reopen it.